domain-knowledge/DOMAIN-OBJECTS.md

```diff
@@ -0,0 +1,501 @@
+# Domain Knowledge and Business Rules
+
+## Terminology
+
+### Service
+
+The "Service" refers collectively to the software and accompanying third-party tools
+that comprise the indieHD suite, _installed and "running" on a server_.
+
+### Service Instance
+
+A "Service Instance" refers to _one particular "installation"_ of the software
+and accompanying third-party tools that comprise the indieHD suite.
+
+### Service Operator
+
+The "Service Operator" refers to the entity that operates the "Service". indieHD, LLC,
+operates a "Service Instance", but is not necessarily the sole, exclusive "Service Operator". 
+
+Given that the indieHD suite is licensed in such a way that any entity may setup a
+"Service Instance" and act as the "Service Operator" thereof, a "Service Operator"
+could therefore be anybody. For this reason, the terms "We" and "Us" are avoided
+in favor of "Service Operator".
+
+**IMPORTANT**: In cases in which indieHD, LLC is _not_ the "Service Operator",
+there is no guarantee that any provision of this document is applicable or accurate,
+as any entity may modify the "Service" in any way, without restriction, provided
+such modifications do not violate the associated copyright and licensing provisions.
+
+## Domain Objects ("Models")
+
+### Account
+
+An `Account` is a merely an extension of a `User`. Whereas a `User` contains the
+bare-minimum information required for an entity to authenticate (email address
+and password), an `Account` contains additional metadata.
+
+This additional metadata includes fields such as name, address, and other
+contact information that is **entirely optional** for the `User` to provide, but that
+would be required under certain circumstances, such as if the account-holder were
+to win a physical prize that requires shipping, such as a t-shirt or CD.
+
+An `Account` cannot exist without a `User`, and vice versa.
+
+**`Account` information is private.**
+
+#### Relationships
+
+- `Account` `belongsTo()` `User`
+- `Account` `belongsTo()` `Country`
+
+#### Access Policy
+
+- Anybody may create an `Account`.
+- An account-holder may view their own `Account`.
+- An account-holder may update their own `Account`.
+- An account-holder may soft-delete their own `Account`.
+
```

Comment:
Maybe we should use the "archive" in place of "soft delete" in the docs? @cbj4074 @poppabear8883 

Unless the docs are really meant for developers.

Reply:
I guess soft delete is fine then.

Reply:
These documents are indeed intended primarily for developers. We'll add a less technical equivalent for end-users.